src/kartograf/atom_mapper.py

```diff
@@ -873,10 +904,46 @@ def suggest_mappings(
         Iterator[AtomMapping]
             returns an interator of possible atom mappings.
         """
-        yield LigandAtomMapping(
-            A,
-            B,
-            self.suggest_mapping_from_rdmols(
-                molA=A.to_rdkit(), molB=B.to_rdkit()
-            ),
-        )
+        # TODO: We probably want to modularize the following in methods
+        if isinstance(A, ProteinComponent) or isinstance(B, ProteinComponent):
+            # 1. identify Component Chains
+            componentA_chains = self._split_protein_components_molecules(A)
+            componentB_chains = self._split_protein_components_molecules(B)
+
+            # 2. calculate all possible mappings
```

Comment:
Do we need to do all possible mappings, should we exclude chains which have a large difference in the number of atoms, to avoid trying to map waters to protein chains this might make it slightly faster?

Reply:
This is great. Are these 10 seconds significant? I don't have data on how much it needed without these changes. 

On the other hand, if we put a hard limit on the number of different atoms, we have the risk of hitting some corner cases. As in, imagine mapping HUGE molecules, where 100 atoms or so is just a small percentage. I'd argue that's probably not the use case for Kartograf, but is there a reason to limit ourselves that way? Other than performance of course. That raises the question, is this a real bottleneck we are experiencing right now? Thanks!

Reply:
As of [cd34743](https://github.com/OpenFreeEnergy/kartograf/pull/47/commits/cd34743b9d964e968a9d5d4405d20e7244d8d7bf) the test now takes 22s so still some room for improvement but I am happy with this for now.